src/health.ts

```diff
@@ -0,0 +1,57 @@
+import fetch from 'node-fetch';
+import { AbortSignal } from 'node-fetch/externals';
+import { URL } from 'url';
+import { KubeConfig } from './config';
+import { RequestOptions } from 'https';
+
+export class Health {
+    public config: KubeConfig;
+
+    public constructor(config: KubeConfig) {
+        this.config = config;
+    }
+
+    public async readyz(opts: RequestOptions): Promise<boolean> {
+        return this.check('/readyz', opts);
+    }
+
+    public async livez(opts: RequestOptions): Promise<boolean> {
+        return this.check('/livez', opts);
+    }
+
+    private async healthz(opts: RequestOptions): Promise<boolean> {
+        return this.check('/healthz', opts);
+    }
+
+    private async check(path: string, opts: RequestOptions): Promise<boolean> {
+        const cluster = this.config.getCurrentCluster();
+        if (!cluster) {
+            throw new Error('No currently active cluster');
+        }
+
+        const requestURL = new URL(cluster.server + path);
+        const requestInit = await this.config.applyToFetchOptions(opts);
+        const controller = new AbortController();
+        requestInit.signal = controller.signal as AbortSignal;
+        requestInit.method = 'GET';
+
+        return await fetch(requestURL.toString(), requestInit)
```

Comment:
Is there any benefit to mixing `async`/`await` style with the `Promises` API? If not, can you use `async`/`await` for simplicity/readability?

Reply:
I got inspiration from `log.ts` which is using this pattern, but no, I cannot see any benefit. I changed to `async`/`await`.